Framework/CurveFitting/src/Algorithms/NormaliseByPeakArea.cpp

```diff
@@ -283,45 +275,38 @@ double NormaliseByPeakArea::fitToMassPeak(const MatrixWorkspace_sptr &yspace,
 }
 
 /**
- * Divides the input Y & E data by the given factor
- * @param area Value to use as normalisation factor
- * @param index Index on input spectrum to normalise
- */
+* Divides the input Y & E data by the given factor
+* @param area Value to use as normalisation factor
+* @param index Index on input spectrum to normalise
+*/
 void NormaliseByPeakArea::normaliseTOFData(const double area,
                                            const size_t index) {
-  const auto &inY = m_inputWS->readY(index);
-  auto &outY = m_normalisedWS->dataY(index);
-  std::transform(inY.begin(), inY.end(), outY.begin(),
-                 std::bind2nd(std::divides<double>(), area));
-
-  const auto &inE = m_inputWS->readE(index);
-  auto &outE = m_normalisedWS->dataE(index);
-  std::transform(inE.begin(), inE.end(), outE.begin(),
-                 std::bind2nd(std::divides<double>(), area));
+  m_normalisedWS->mutableY(index) = m_inputWS->y(index) / area;
+  m_normalisedWS->mutableE(index) = m_inputWS->e(index) / area;
 }
 
 /**
- * @param accumWS Workspace used to accumulate the final data
- * @param yValues Input signal values for y-space
- * @param eValues Input errors values for y-space
- * @param index Index of the workspace. Only used when not summing.
- */
+* @param accumWS Workspace used to accumulate the final data
+* @param yValues Input signal values for y-space
+* @param eValues Input errors values for y-space
+* @param index Index of the workspace. Only used when not summing.
+*/
 void NormaliseByPeakArea::saveToOutput(const API::MatrixWorkspace_sptr &accumWS,
-                                       const std::vector<double> &yValues,
-                                       const std::vector<double> &eValues,
+                                       const HistogramData::HistogramY &yValues,
+                                       const HistogramData::HistogramE &eValues,
```

Comment:
Look at how this function is called, and how Y and E are used below --- at least sometimes data could be shared, so consider passing shared pointers here, i.e., the result of `sharedY` etc, and then use `setSharedY` below.